lib/std/fmt.zig

```diff
@@ -179,7 +180,7 @@ pub fn format(
         };
 
         const arg_to_print = comptime arg_state.nextArg(arg_pos) orelse
-            @compileError("too few arguments");
+            @compileError(comptimePrint("too few arguments in '{}'", .{fmtEscapes(fmt)}));
```

Comment:
This is too crazy, calling formatted printing inside the error message for bad formatting printing, which is all just to work around a limitation of compiler error messages when there is a comptime call stack.

It reeks of C++.

This is a great example of climbing up the wrong mountain.

Instead let us address the real problems:
- compile errors for formatted printing are not good enough *because of the compiler's inadequacy, not the std lib*
- `@compileError` should be the one to render unprintables correctly
- some kind of comptime-equivalent of `@returnAddress()` is needed

Reply:
you have a great point, the thought absolutely crossed my mind when making this pr, this *shouldnt* be necessary. the problem is that, as of now, i believe it is necessary. zig in general is not known for having friendly or descriptive error messages and `std.fmt` is notorious for bad ones. unless someone takes it upon themselves to improve the compiler enough to supersede this pr soon i think it would be a mistake not having it in some form or another. whats the path forward?

Reply:
Also related https://github.com/ziglang/zig/issues/12093#issuecomment-1585340480

Reply:
something like that would be wonderful, one limitation of my pr here is that it doesnt have access to the actual in-source format string so if someone for example did `print("{}\x0a", .{})` my error would report back `"{}\n"` since `\x0a` is the hex representation of a line feed